crates/sui/src/sui_commands.rs

```diff
@@ -631,7 +631,7 @@ async fn start(
         swarm_builder = swarm_builder.with_epoch_duration_ms(epoch_duration_ms);
     } else {
         if config.is_none() && !sui_config_dir()?.join(SUI_NETWORK_CONFIG).exists() {
-            genesis(None, None, None, false, epoch_duration_ms, None, false).await?;
+            genesis(None, None, None, false, epoch_duration_ms, None, false).await.map_err(|_| anyhow!("Cannot run genesis with non-empty Sui config directory: {}.\n\nIf you are trying to run a local network without persisting the data (so a new genesis that is randomly generated and will not be saved once the network is shut down), use --force-regenesis flag.\nIf you are trying to persist the network data and start from a new genesis, use sui genesis --help to see how to generate a new genesis.", sui_config_dir().unwrap().display()))?;
```

Comment:
What's the error message this used to produce? Usually it's better to use `context(...)` to add extra information to the existing error message rather than entirely replace it, but maybe the existing error message doesn't help here.

Reply:
The error message that would show is the one for genesis command. The issue is that genesis command has some extra flags (e.g., --force), which we should not add here, so when that error message gets propagated here it does not make sense. 
I think it's better to provide a specific error message here given this situation.
For some reason that I do not know, I've never been a fan of `context` 😅

Reply:
I think in a CLI setting it may not be great, it tends to work pretty well for errors coming from a service, where you can want to share a logical "call stack".